tests/common/dualtor/dual_tor_utils.py

```diff
@@ -636,6 +639,152 @@ def check_tunnel_balance(ptfhost, standby_tor_mac, vlan_mac, active_tor_ip, stan
                socket_recv_size=16384)
 
 
+def generate_hashed_packet_to_server(ptfadapter, duthost, hash_key, target_server_ip):
+    """
+    Generate a packet to server based on hash.
+    The value of field in packet is filled with random value according to hash_key
+    """
+    SRC_IP_RANGE = [unicode('8.0.0.0'), unicode('8.255.255.255')]
+    base_src_mac = ptfadapter.dataplane.get_mac(0, 0)
+    ip_src = random_ip(SRC_IP_RANGE[0], SRC_IP_RANGE[1]) if hash_key == 'src-ip' else SRC_IP_RANGE[0]
+    ip_dst = target_server_ip
+    sport = random.randint(1, 65535) if hash_key == 'src-port' else 1234
+    dport = random.randint(1, 65535) if hash_key == 'dst-port' else 80
+    src_mac = (base_src_mac[:-5] + "%02x" % random.randint(0, 255) + ":" + "%02x" % random.randint(0, 255)) if hash_key == 'src-mac' else base_src_mac
+    dst_mac = duthost.facts["router_mac"]
+    vlan_id = random.randint(1, 4094) if hash_key == 'vlan-id' else 0
+    pkt = testutils.simple_tcp_packet(pktlen=128 if vlan_id == 0 else 132,
+                        eth_dst=dst_mac,
+                        eth_src=src_mac,
+                        dl_vlan_enable=False if vlan_id == 0 else True,
+                        vlan_vid=vlan_id,
+                        vlan_pcp=0,
+                        ip_src=ip_src,
+                        ip_dst=ip_dst,
+                        tcp_sport=sport,
+                        tcp_dport=dport,
+                        ip_ttl=64)
+    exp_pkt = mask.Mask(pkt)
+    exp_pkt.set_do_not_care_scapy(scapyall.Ether, "dst")
+    exp_pkt.set_do_not_care_scapy(scapyall.Ether, "src")
+    exp_pkt.set_do_not_care_scapy(scapyall.IP, "tos")
+    exp_pkt.set_do_not_care_scapy(scapyall.IP, "ttl")
```

Comment:
Why ignore almost all the fields. What is being checked in the expected packet? Is it because of the encapsulation?

Reply:
Yes, it is to put a common match for packets sent to downlinks and uplinks. I added IP src/dst match now.
Are there any other fields that I can remove the ignore flag for?